packages/gatsby-plugin-utils/src/polyfill-remote-file/types.ts

```diff
@@ -93,3 +93,33 @@ export function isImage(node: {
 
   return node.mimeType.startsWith(`image/`) && node.mimeType !== `image/svg+xml`
 }
+
+export type ImageCdnTransformArgs = WidthOrHeight & {
+  format: string
+  cropFocus?: ImageCropFocus | Array<ImageCropFocus>
+  quality: number
+}
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type ImageCdnSourceImage = {
+  url: string
+  mimeType: string
+  filename: string
+  internal: { contentDigest: string }
+}
+
+/**
+ * The function is used to optimize image delivery by generating URLs that leverage CDN capabilities
+ * @param {ImageCdnSourceImage} source - An object representing the source image, including properties like
+ * URL, filename, and MIME type.
+ * @param {ImageCdnTransformArgs} imageArgs - An object containing arguments for image transformation, such as
+ * format, quality, and crop focus.
+ * @param {string} pathPrefix - A string representing the path prefix to be prepended to the
```

Comment:
Note here - `pathPrefix` is ideally prepended just so image transformation endpoint is "situated" in the root of the site, but it doesn't actually have to be used. 

As-is in `gatsby-adapter-netlify` it actually is NOT used as to my knowledge `/.netlify/images` needs to be in domain root (likely this could be handled with rewrite / redirect like `<pathPrefix>/.netlify/images` -> `/.netlify/images`, but IMO avoiding extra step in request/response pipeline is better than making it situated in root - this is passed only in case there is endpoint is easily configurable)

Reply:
Right its not used by our Netlify implementation, so maybe there's a better way to describe this being optionally needed to generate the needed cdn url.

Reply:
Maybe something like this is better?
`@param {string} pathPrefix - This parameter allows for an optional path prefix in the generated URL, primarily influencing the location of the image transformation endpoint, if not in the domain root.`

Reply:
Used suggestion in https://github.com/gatsbyjs/gatsby/pull/38685/commits/c7b3da37c6b54ed456b85f1ba4f8e022ed88c136